server/src/main/java/io/crate/statistics/TransportAnalyzeAction.java

```diff
@@ -100,8 +102,10 @@ public TransportAnalyzeAction(TransportService transportService,
             FetchSampleRequest::new,
             // Explicit generic is required for eclipse JDT, otherwise it won't compile
             new NodeActionRequestHandler<FetchSampleRequest, FetchSampleResponse>(
-                req -> completedFuture(new FetchSampleResponse(
-                    reservoirSampler.getSamples(req.relation(), req.columns(), req.maxSamples())))
+                req -> {
+                    runningAnalyze.compareAndExchange(null, reservoirSampler.getSamples(req.relation(), req.columns(), req.maxSamples()));
```

Comment:
`runningAnalyze.compareAndExchange` + `runningAnalyze.get` is not atomic but I think should be fine(?) as long as first line is atomic - even if another `compareAndExchange` squeezes in between it will take non-null value of the first `compareAndExchange`